### DISCUSSION TITLE
post-inference tracking error on project with multiple videos

### DISCUSSION THREAD

**Discussion by donovanashby** [talmolab/sleap#1813] [Help!] [answered]
Hi,
I have a trained model I am happy with, in a project with 30+ videos. I have successfully run inference on all videos and everything looks good (all done in the GUI). I'm using CLI to run post-inference tracking, but I am getting errors when I try to run the whole project with the following command:

`sleap-track --tracking.tracker simplemaxtracks --tracking.max_tracking 1 --tracking.max_tracks 2 --tracking.similarity centroid --tracking.match hungarian --tracking.track_window 2 --tracking.target_instance_count 2 --tracking.post_connect_single_breaks 1 -o "project.tracked.slp"  "project.untracked.slp"`

I see the following error:

> Traceback (most recent call last):
>   File "/home/ali/Apps/anaconda3/envs/sleap/bin/sleap-track", line 33, in <module>
>     sys.exit(load_entry_point('sleap==1.3.3', 'console_scripts', 'sleap-track')())
>   File "/home/ali/Apps/anaconda3/envs/sleap/lib/python3.7/site-packages/sleap/nn/inference.py", line 5439, in main
>     frames = run_tracker(frames=frames, tracker=tracker)
>   File "/home/ali/Apps/anaconda3/envs/sleap/lib/python3.7/site-packages/sleap/nn/tracking.py", line 1456, in run_tracker
>     instances=tracker.track(**track_args),
>   File "/home/ali/Apps/anaconda3/envs/sleap/lib/python3.7/site-packages/sleap/nn/tracking.py", line 715, in track
>     self.update_matched_instance_tracks(frame_matches.matches)
>   File "/home/ali/Apps/anaconda3/envs/sleap/lib/python3.7/site-packages/sleap/nn/tracking.py", line 759, in update_matched_instance_tracks
>     tracking_score=match.score,
>   File "/home/ali/Apps/anaconda3/envs/sleap/lib/python3.7/site-packages/attr/_funcs.py", line 370, in evolve
>     return cls(**changes)
> TypeError: __init__() got an unexpected keyword argument 'tracking_score'

I don't get this error when I run an individual prediction file for one video from the project, just the whole project. On an individual file it seems to process file and give me a tracked file. I'm running everything locally.

**Comment by gitttt-1234** [comment, marked as answer]
Since this was posted, we've updated the tracking pipeline and moved all training, inference, and tracking to [**sleap-nn**](https://nn.sleap.ai/latest/), which is now the backend SLEAP uses for deep learning. For post-inference tracking specifically, check out the [tracking guide](https://nn.sleap.ai/latest/guides/tracking/) which covers the equivalent CLI workflow with the new package.

Closing in favor of the updated tracking in sleap-nn. If you run into anything with the new pipeline, feel free to open a new discussion!
